graylog2-server/src/main/java/org/graylog2/plugin/Message.java

```diff
@@ -448,6 +548,10 @@ public void addStreams(Iterable<Stream> newStreams) {
      */
     public boolean removeStream(Stream stream) {
         final boolean removed = streams.remove(stream);
+        sizeCounter.dec(8);
```

Comment:
We should only decrement the size counter if the stream has actually been removed from the set. (check `stream.remove(stream)` return value. (check already done in line 556)